smapply/tasks_test.py

```diff
@@ -53,3 +54,20 @@ def test_sync_all_users_bad_data(mocker):
     )
     sync_all_users()
     assert Profile.objects.count() == 1
+
+
+def test_sync_existing_user_lacking_data(mocker):
+    """
+    Test that an already existing profile without smapply_user_data has that data populated during sync_all_users.
+    """
+    profile = ProfileFactory.create(smapply_id=12345678)
+
+    mocker.patch(
+        'smapply.tasks.list_users',
+        return_value=test_user_data,
+    )
+    sync_all_users()
+    assert Profile.objects.count() == 2
+
+    profile.refresh_from_db()
+    assert profile.smapply_user_data
```

Comment:
```python
@pytest.mark.parametrize("sma_data", [None, {"foo": "bar"}])
def test_sync_existing_user_data(mocker, sma_data):
    """
    Test that an existing profile with and without smapply_user_data has data populated during sync_all_users.
    """
    profile = ProfileFactory.create(smapply_id=12345678, smapply_user_data=sma_data)

    mocker.patch(
        'smapply.tasks.list_users',
        return_value=test_user_data,
    )
    sync_all_users()

    profile.refresh_from_db()
    assert profile.smapply_user_data == test_user_data[0]
```

Reply:
Added

Reply:
You can remove the above test (`test_sync_existing_user_lacking_data`), it should test the same conditions.

Reply:
Okay